Contract tests refactor open

loredanacirstea opened the pull request:
To be reviewed after https://github.com/raiden-network/microraiden/pull/273 is merged.

- move create channel tests from the `get_channel` fixture used in all tests to dedicated tests
- shortens travis time relative to the above PR

LefterisJP commented on contracts/tests/test_channel_create.py:
```diff
@@ -115,6 +117,56 @@ def test_channel_223_create_bounty_limit(get_block, owner, get_accounts, uraiden
+    token = token_instance
+    (sender, receiver) = get_accounts(2)
+
+    # Make sure you have a fixture with a supply > 2 ** 192 + 100
```
Shouldn't you put an `assert` instead of a comment here?

loredanacirstea replied:
I was using an array of contract parameters to run the tests multiple times. But this is an edge case, so I will handle it only in this test with clear parameters. Refactor on the way.

LefterisJP commented on contracts/tests/test_channel_create.py:
```diff
@@ -115,6 +117,56 @@ def test_channel_223_create_bounty_limit(get_block, owner, get_accounts, uraiden
+            )
+
+
+def test_channel_223_event(owner, get_accounts, uraiden_instance, token_instance, event_handler, print_gas):
```
what is channel 223? Is it ERC223? Then please specify.

LefterisJP commented on contracts/tests/test_channel_create.py:
```diff
@@ -115,6 +117,56 @@ def test_channel_223_create_bounty_limit(get_block, owner, get_accounts, uraiden
+    )
+    ev_handler.check()
+
+    print_gas(txn_hash, 'channel_223_create')
```
the print gas is for test debugging purposes?

loredanacirstea replied:
Yes, quick way to have some values after testing.

LefterisJP commented on contracts/tests/test_channel_create.py:
```diff
@@ -190,28 +242,80 @@ def test_channel_20_create_bounty_limit(owner, get_accounts, uraiden_instance, t
-    get_accounts,
-    uraiden_instance,
-    token_instance):
+def test_channel_20_event(owner, get_accounts, uraiden_instance, token_instance, event_handler, txn_gas, print_gas):
```
what is channel 20 event? Is t ERC20 then please specify :P

loredanacirstea commented:
Refactored the uint256 -> uint192 test , but I left one positive test commented due the deposit bug bounty limit. I'll open an issue to not forget about it. The tests above it are more important, so it doesn't make sense now to find complicated ways to add this one.

Also added a TODO in the code:
https://github.com/loredanacirstea/microraiden/blob/3b0e7f1c8c207b08a0e2214ed04fe842b238725d/contracts/tests/test_channel_create.py#L154
